src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -296,9 +298,57 @@ class DashboardDetailsSerializer(CamelSnakeSerializer):
     id = serializers.CharField(required=False)
     title = serializers.CharField(required=False, max_length=255)
     widgets = DashboardWidgetSerializer(many=True, required=False)
+    projects = ListField(child=serializers.IntegerField(), required=False, default=[])
+    environment = ListField(child=serializers.CharField(), required=False, allow_null=True)
+    range = serializers.CharField(required=False, allow_null=True)
+    start = serializers.DateTimeField(required=False, allow_null=True)
+    end = serializers.DateTimeField(required=False, allow_null=True)
+    filters = serializers.DictField(required=False)
 
     validate_id = validate_id
 
+    def validate_projects(self, projects):
+        projects = set(projects)
+
+        # Don't need to check all projects or my projects
+        if projects == ALL_ACCESS_PROJECTS or len(projects) == 0:
+            return projects
+
+        # Check that there aren't projects in the query the user doesn't have access to
+        if len(projects - {project.id for project in self.context["projects"]}) > 0:
+            raise PermissionDenied
+
+        return projects
```

Comment:
If this is replicated logic from the Discover Saved Query should we pull it out into a function and reuse?

Reply:
Pulled it out into a function called `validate_project_ids`, but when running the tests I kept finding an import error:
```
ImportError: cannot import name 'validate_project_ids' from partially initialized module 'sentry.api.validators' (most likely due to a circular import)
```

so I moved the import into the function call. It's a solution I've seen before when there are a bunch helpers being exported through a `__init__.py` like we have in `sentry/api/validators/__init__.py`